src/librustdoc/html/static/js/search.js

```diff
@@ -587,7 +597,7 @@ window.initSearch = function(rawSearchIndex) {
      *
      * The supported syntax by this parser is as follow:
      *
-     * ident = *(ALPHA / DIGIT / "_")
+     * ident = *(ALPHA / DIGIT / "_") [!]
```

Comment:
This doesn't quite match the actual implemented rule.

According to this eBNF, `!::!` is valid, because there's one bang at the end of each ident. But the actual search.js rule allows only one `!` in the entire *path*.

Reply:
Indeed! I need to update the eBNF.

Reply:
Hum actually, should we allow multiple `!` in a path? Like `!::!`?

Reply:
It doesn't actually matter to me either way.

It shouldn't ever match anything, because we can work it out for the four possible ways you can get paths that look like that, and they're all impossible:

* `never::macro`: macros can currently only appear in modules, not primitives
* `never::never`: the Never primitive has no parent item
* `macro::macro`: macros have no child items
* `macro::never`: macros have no child items

However, I'm not sure we need to reject it in the parser. All that really matters to me is that the parser reject things that look like `!foo`, for future-proofing purposes. We might want to use that for exclusion syntax later (a NOT operator, basically).

Reply:
Ok, let's support `!::!` then.